File: python/examples/upload_file.py
#!/usr/bin/env python3
"""Usage: AHOLO_API_KEY=xxx python upload_file.py ./photo.jpg"""
import os
import sys
import time

from manycore.aholo_sdk_asset import create_asset_client
from manycore.aholo_sdk_core import AholoClientConfig

if len(sys.argv) < 2:
    raise SystemExit("Usage: AHOLO_API_KEY=xxx python upload_file.py <file-path>")

file_path = sys.argv[1]
region = os.environ.get("AHOLO_REGION", "cn")
asset = create_asset_client(AholoClientConfig(region=region))

def show_progress(uploaded: int, total: int) -> None:
    pct = int(uploaded / total * 100)
    print(f"\r  {pct}% ({uploaded / 1024 / 1024:.1f}/{total / 1024 / 1024:.1f} MB)", end="", flush=True)

print(f"Uploading {file_path} ...")
t0 = time.monotonic()
result = asset.upload_file(file_path, on_progress=show_progress)
print()  # newline after progress bar
elapsed_ms = int((time.monotonic() - t0) * 1000)
print(f"Upload complete ({elapsed_ms}ms)")
print(f"url: {result.url}")

File: java/examples/src/main/java/com/manycoreapis/sdk/examples/UploadFile.java
package com.manycoreapis.sdk.examples;

import com.manycoreapis.sdk.asset.AssetClient;
import com.manycoreapis.sdk.asset.UploadResult;
import com.manycoreapis.sdk.core.AholoClientConfig;

import java.nio.file.Paths;

/**
 * Minimal example: upload a local file and print the public URL.
 *
 * Usage:
 *   AHOLO_API_KEY=xxx mvn exec:java -pl examples \
 *     -Dexec.mainClass=com.manycoreapis.sdk.examples.UploadFile \
 *     -Dexec.args="/path/to/photo.jpg"
 */
public class UploadFile {
    public static void main(String[] args) throws Exception {
        if (args.length < 1) {
            System.err.println("Usage: UploadFile <file-path>");
            System.exit(1);
        }

        String filePath = args[0];
        String region = System.getenv().getOrDefault("AHOLO_REGION", "cn");
        AssetClient asset = AssetClient.create(AholoClientConfig.ofRegion(region));

        System.out.println("Uploading " + filePath + " ...");
        long t0 = System.currentTimeMillis();

        UploadResult result = asset.uploadFile(Paths.get(filePath), (uploaded, total) -> {
            int pct = (int) (uploaded * 100 / total);
            System.out.printf("\r  %d%% (%.1f/%.1f MB)", pct,
                    uploaded / 1024.0 / 1024.0, total / 1024.0 / 1024.0);
            System.out.flush();
        });
        System.out.println();

        System.out.printf("Upload complete (%dms)%n", System.currentTimeMillis() - t0);
        System.out.println("url: " + result.url());
    }
}
